packages/hoppscotch-cli/src/utils/mutators.ts

```diff
@@ -67,13 +70,34 @@ export async function parseCollectionData(
     ? contents
     : [contents];
 
-  if (maybeArrayOfCollections.some((x) => !isRESTCollection(x))) {
+  const collectionSchemaParsedResult = z
+    .array(entityReference(HoppCollection))
+    .safeParse(maybeArrayOfCollections);
+
+  if (!collectionSchemaParsedResult.success) {
     throw error({
       code: "MALFORMED_COLLECTION",
       path,
       data: "Please check the collection data.",
     });
   }
 
-  return maybeArrayOfCollections as HoppCollection[];
+  return collectionSchemaParsedResult.data.map((collection) => {
+    const requestSchemaParsedResult = z
+      .array(entityReference(HoppRESTRequest))
+      .safeParse(collection.requests);
+
+    if (!requestSchemaParsedResult.success) {
+      throw error({
+        code: "MALFORMED_COLLECTION",
+        path,
+        data: "Please check the collection data.",
+      });
+    }
+
+    return {
+      ...collection,
+      requests: requestSchemaParsedResult.data,
+    };
```

Comment:
Schema validation is performed for each request in the collection against the `HoppRESTRequest` versioned entity to ensure requests conforming to an old version are translated to the latest version. This isn't achievable at the `HoppCollection` validation phase since the translation function only deals with the top-level fields of the collection.